ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -674,7 +674,9 @@ void wasSuccessful() {
     @Override public void close() throws Exception {
       try {
         //the transaction is about to close, we can stop heartbeating regardless of it's state
-        CompactionHeartbeatService.getInstance(conf).stopHeartbeat(txnId);
+        if (txnId > 0) {
```

Comment:
I think we should simply move the following check at the beginning of the close() method  
````
      if (status == TxnStatus.UNKNOWN) {
        return;
      }
````